src/aws/auth.rs

```diff
@@ -184,6 +272,43 @@ mod tests {
         assert!(matches!(config.auth, AwsAuthentication::Role { .. }));
     }
 
+    #[test]
+    fn parsing_assume_role_with_imds_client() {
+        let config = toml::from_str::<ComponentConfig>(
+            r#"
+            auth.assume_role = "root"
+            auth.imds.max_attempts = 5
+            auth.imds.connect_timeout_seconds = 30
+            auth.imds.read_timeout_seconds = 10
+        "#,
+        )
+        .unwrap();
+
+        const CONNECT_TIMEOUT: Duration = Duration::from_secs(30);
+        const READ_TIMEOUT: Duration = Duration::from_secs(10);
```

Comment:
```suggestion
```

Think these can be deleted since you pulled them up and out of the individual test function?

Reply:
good catch